packages/agreement-process/test/db.test.ts

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable functional/no-let */
+/* eslint-disable functional/immutable-data */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import { afterEach, beforeAll, describe, expect, it } from "vitest";
+import {
+  AgreementCollection,
+  EServiceCollection,
+  ReadModelRepository,
+  TenantCollection,
+  getMongodbContainer,
+  getPostgreSqlContainer,
+  initDB,
+} from "pagopa-interop-commons";
+import { IDatabase } from "pg-promise";
+import { config } from "../src/utilities/config.js";
+import { readModelServiceBuilder } from "../src/services/readModelService.js";
+import { agreementServiceBuilder } from "../src/services/agreementService.js";
+
+describe("database test", async () => {
+  let agreements: AgreementCollection;
+  let eservices: EServiceCollection;
+  let tenants: TenantCollection;
+  let readModelService;
+  let agreementService;
+  let postgresDB: IDatabase<unknown>;
+
+  beforeAll(async () => {
+    const postgreSqlContainer = await getPostgreSqlContainer({
+      dbName: config.eventStoreDbName,
+      username: config.eventStoreDbUsername,
+      password: config.eventStoreDbPassword,
+    }).start();
+
+    const mongodbContainer = await getMongodbContainer({
+      dbName: config.readModelDbName,
+      username: config.readModelDbUsername,
+      password: config.readModelDbPassword,
+    }).start();
+
+    config.eventStoreDbPort = postgreSqlContainer.getMappedPort(5432);
+    config.readModelDbPort = mongodbContainer.getMappedPort(27017);
+
+    const readModelRepository = ReadModelRepository.init(config);
+    agreements = readModelRepository.agreements;
+    eservices = readModelRepository.eservices;
+    tenants = readModelRepository.tenants;
+
+    readModelService = readModelServiceBuilder(config);
+    agreementService = agreementServiceBuilder(config);
+
+    postgresDB = initDB({
+      username: config.eventStoreDbUsername,
+      password: config.eventStoreDbPassword,
+      host: config.eventStoreDbHost,
+      port: config.eventStoreDbPort,
+      database: config.eventStoreDbName,
+      schema: config.eventStoreDbSchema,
+      useSSL: config.eventStoreDbUseSSL,
+    });
+  });
+
+  afterEach(async () => {
+    await agreements.deleteMany({});
+    await eservices.deleteMany({});
+    await tenants.deleteMany({});
+
+    await postgresDB.none("TRUNCATE TABLE agreement.events RESTART IDENTITY");
+    await postgresDB.none("TRUNCATE TABLE catalog.events RESTART IDENTITY");
+    // await postgresDB.none("TRUNCATE TABLE tenant.events RESTART IDENTITY");
```

Comment:
```suggestion

```

Reply:
I wasn't sure if the agreement tests will also require tenants. We can remove it later if not needed.

Reply:
I wasn't sure if the tenants are required during the `agreement` tests. We can add that line again if we use that table during this test suite.